core/hydromet_stratified.py

```diff
@@ -0,0 +1,560 @@
+
+
+def Avg_R(lower_bound: float, upper_bound: float, GEV_parameters: np.ndarray, PMP: float) -> float:
+    """Calculates the average value of the GEV distribution of rainfall or runoff based on an upper and lower bound.
+    """
+    return quad(Avg_R_integrand, lower_bound, upper_bound, args=(GEV_parameters, PMP))
+
+
+def GEV_RI(RI: np.ndarray, GEV_parameters: np.ndarray, PMP: float) -> np.ndarray:
+    """Provides rainfall or runoff as a function of the return interval (RI).
+    """
+    return PPF_GEV(1-1.0/RI, GEV_parameters, PMP)
+
+
+def objective_func_bound_GEV(RI_lower: float, RI_upper: float, RI_middle: float, GEV_parameters: np.ndarray, 
+                             PMP: float) -> float:
+    """Calculates the square of the error between the average rainfall or runoff calculated from the bin floor and 
+       ceiling (given in terms of RI) and the rainfall or runoff of the return period of interest.
+    """ 
+    return np.square(Avg_R(GEV_RI(RI_lower, GEV_parameters, PMP), GEV_RI(RI_upper, GEV_parameters, PMP), GEV_parameters, 
+                                  PMP)[0]/(1.0/RI_lower-1.0/RI_upper) - GEV_RI(RI_middle, GEV_parameters, PMP))
+
+
+def bound_lower_GEV(RI_upper: float, RI_middle: float, GEV_parameters: np.ndarray, initial_value: float,
+                    PMP: float) -> float:
+    """Finds the rainfall or runoff bin floor given the bin ceiling and average return interval of the bin.
+    """
+    return minimize(objective_func_bound_GEV, initial_value, args = (RI_upper, RI_middle, GEV_parameters, PMP),
+                    method='SLSQP', bounds=[(1.0, RI_upper*0.999)], options={'disp': False})
+
+
+def PDF_QlS(Q: np.ndarray, S: float, mu: float, GEV_parameters: np.ndarray, PMP: float) -> np.ndarray:
+    """This function provides the runoff PDF conditional on max potential retention, where Q=runoff, S=max potential 
+       retention, and mu=initial abstraction parameter.
+    """
+    return (Q+2.0*S+np.sqrt(Q)*np.sqrt(Q+4.0*S))/(2.0*np.sqrt(Q)*np.sqrt(Q+4.0*S))*\
+            PDF_GEV(1.0/2.0*(Q+np.sqrt(Q)*np.sqrt(Q+4.0*S)+2.0*S*mu), GEV_parameters, PMP)
+
+
+def PDF_Q(Q: float, mu: float, GEV_parameters: np.ndarray, PMP: float, partition_avg: np.ndarray,
+          Delta_P: float, error_PQ: float) -> float: 
+    """
+    """
+    return sum(Delta_P*PDF_QlS(Q, S_avg_partition, mu, GEV_parameters, PMP) for S_avg_partition in 
+               partition_avg)/(1-error_PQ)
+
+
+def Qzero_integrand(S: float, mu: float, alpha: float, beta: float, S_limit: float, GEV_parameters: np.ndarray,
+                    PMP: float, error_PQ: float) -> float:
+    """Defines the integrand for calculating the probability of zero runoff.
+    """
+    return (CDF_GEV(S*mu, GEV_parameters, PMP)-CDF_GEV(0, GEV_parameters, PMP))\
+           *(1.0/S_limit)*stats.beta(alpha, beta).pdf(S/S_limit)/(1-error_PQ)
+
+
+def P_Qzero(mu: float, alpha: float, beta: float, S_limit: float, GEV_parameters: np.ndarray, PMP: float, 
```

Comment:
General comment: where there is a parameter like S_Limit, please add a descriptor and an example value or range of acceptable values.